### PR TITLE
Adds initial code

### PR DESCRIPTION
**Pull request by johnbradley**
Adds `dm_formatter.py` script containing @thibaulttabarin's Minnows specific metadata transformation code.
Adds a `Dockerfile` and GitHub Action config to build on release.
Adds a test script to exercise the metadata transformation code and GitHub Action config to run the tests.

### CODE REVIEW

File: dm_formatter.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Convert drexel_metadata JSON file to BGNN minnow metadata JSON file
+import json
+import argparse
+
+
+def reformat_for_bgnn(result):
+    """
+    Reformat and reduce the size of the result dictionary. 
+    Collect only the data necessary for BGNN minnow project. The new format matches the 
+    BGNN_metadata version. Therefore some of the value not calcualted in drexel version are by
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
    BGNN_metadata version. Therefore some of the values not calculated in drexel version are by 
```

File: dm_formatter.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import json
+import argparse
+
+
+def reformat_for_bgnn(result):
+    """
+    Reformat and reduce the size of the result dictionary. 
+    Collect only the data necessary for BGNN minnow project. The new format matches the 
+    BGNN_metadata version. Therefore some of the value not calcualted in drexel version are by 
+    defaulset to "None".
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
    default set to "None". 
```

File: dm_formatter.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    """
+    
+    name_base = list(result.keys())[0]
+    first_value = list(result.values())[0]
+    
+    # Fish metadata
+    fish_num = first_value['fish_count']
+    fish_bbox = first_value['fish'][0]['bbox']
+    pixel_analysis = False if first_value['fish'][0]['pixel_analysis_failed'] else True
```
Comment (hlapp) [inline dm_formatter.py]: Wouldn't it be simpler to just use this assignment?
```suggestion
    pixel_analysis = !first_value['fish'][0]['pixel_analysis_failed']
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FROM python:3.8.10-slim-buster
+
+WORKDIR /pipeline
+
+COPY dm_formatter.py /pipeline
+CMD echo "python dm_formatter.py [input] [output]"
```
Comment (hlapp) [inline Dockerfile]: This makes it look like input and output are optional, which it doesn't seem they actually are, and it also differs from the Usage section in the README.

File: dm_formatter.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    if first_value['fish'][0]['has_eye']:
+        eye_center = first_value['fish'][0]['eye_center']
+    else :
+        eye_center = "None"
+    
+    eye_direction = first_value['fish'][0]['side']
+    foreground_mean = first_value['fish'][0]['foreground']['mean']
+    background_mean = first_value['fish'][0]['background']['mean']
+        
+    dict_fish = {'fish_num': fish_num,"bbox":fish_bbox,
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
    dict_fish = {'fish_num': fish_num,"bbox": fish_bbox, 
```

File: dm_formatter.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        eye_center = first_value['fish'][0]['eye_center']
+    else :
+        eye_center = "None"
+    
+    eye_direction = first_value['fish'][0]['side']
+    foreground_mean = first_value['fish'][0]['foreground']['mean']
+    background_mean = first_value['fish'][0]['background']['mean']
+        
+    dict_fish = {'fish_num': fish_num,"bbox":fish_bbox, 
+                 'pixel_analysis':pixel_analysis, 'rescale':"None",
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
                 'pixel_analysis': pixel_analysis, 'rescale': "None", 
```

File: dm_formatter.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    else :
+        eye_center = "None"
+    
+    eye_direction = first_value['fish'][0]['side']
+    foreground_mean = first_value['fish'][0]['foreground']['mean']
+    background_mean = first_value['fish'][0]['background']['mean']
+        
+    dict_fish = {'fish_num': fish_num,"bbox":fish_bbox, 
+                 'pixel_analysis':pixel_analysis, 'rescale':"None", 
+                 'eye_bbox': "None", 'eye_center':eye_center , 'angle_degree': "None",
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
                 'eye_bbox': "None", 'eye_center': eye_center , 'angle_degree': "None",
```

File: dm_formatter.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                 'eye_direction':eye_direction, 'foreground_mean':round(foreground_mean,2),
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
                 'eye_direction': eye_direction, 'foreground_mean': round(foreground_mean,2), 
```

File: dm_formatter.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    
+    eye_direction = first_value['fish'][0]['side']
+    foreground_mean = first_value['fish'][0]['foreground']['mean']
+    background_mean = first_value['fish'][0]['background']['mean']
+        
+    dict_fish = {'fish_num': fish_num,"bbox":fish_bbox, 
+                 'pixel_analysis':pixel_analysis, 'rescale':"None", 
+                 'eye_bbox': "None", 'eye_center':eye_center , 'angle_degree': "None",
+                 'eye_direction':eye_direction, 'foreground_mean':round(foreground_mean,2), 
+                 'background_mean':round(background_mean,2)}
```
Comment (hlapp) [inline dm_formatter.py]:
```suggestion
                 'background_mean': round(background_mean,2)}
```

**Review by hlapp — APPROVED** [review]
Comments and suggestions are all very minor, so I'm approving in advance.

File: dm_formatter.py (hunk 3 above)
Comment (johnbradley) [reply dm_formatter.py]: FYI: This should have been `pixel_analysis = not first_value['fish'][0]['pixel_analysis_failed']`.

**Comment by johnbradley** [comment]
@thibaulttabarin Do you have any feedback for this PR?

**Comment by thibaulttabarin** [comment]
> 

I didn't have time to do it throughly, it looks good but I want to test everything and i didn't have to finish to look at the drexel_metadata.

**Review by thibaulttabarin — CHANGES_REQUESTED** [review]
add 
#!/usr/bin/env python3

**Review by thibaulttabarin — CHANGES_REQUESTED** [review]
add 

# ADD scripts in /pipeline to the PATH
ENV PATH="/pipeline:${PATH}"

**Comment by johnbradley** [comment]
@thibaulttabarin I added changes to put dm_formatter.py into the PATH and added the shebang.
If you have a metadata file named `INHS_FISH_74711_metadata.json` in the current directory you should be able to test out the changes like so:
```
docker build -t dmf .
docker run -v $(pwd):/data -it dmf dm_formatter.py /data/INHS_FISH_74711_metadata.json /data/result.json
```

**Review by thibaulttabarin — APPROVED** [review]

**Pull request merged**